webservices/legal_docs/advisory_opinions.py

```diff
@@ -58,18 +58,24 @@
     WHERE ao_id = %s
 """
 
-STATUTE_CITATION_REGEX = re.compile(r"(?P<title>\d+)\s+U.S.C.\s+§*(?P<section>\d+).*\.?")
-REGULATION_CITATION_REGEX = re.compile(r"(?P<title>\d+)\s+CFR\s+§*(?P<part>\d+)\.(?P<section>\d+)")
-AO_CITATION_REGEX = re.compile(r"\b(?P<year>\d{4,4})-(?P<serial_no>\d+)\b")
+STATUTE_CITATION_REGEX = re.compile(
+    r"(?P<title>\d+)\s+U.S.C.\s+§*\s*(?P<section>\d+).*\.?")
```

Comment:
I think you need to escape the `.` in the `U.S.C.` section, so that the regex looks like this:

`r"(?P<title>\d+)\s+U\.S\.C\.\s+§*\s*(?P<section>\d+).*\.?"`

You have this below in the citation regex, which is what you want!  What happens without escaping them is that you will still match a pattern that has different characters, e.g., `52 UxSxCx § 30101`.

Additionally, do you also need to be able to match `USC` without the `.` between each character?  If so, you'll need the extra `*` character after each `\.` like you have below in the citation regex.

Reply:
Thanks, @ccostino! It seems like a good idea to me, but I didn't actually touch that part myself. @vrajmohan - do you see any problem with adding this to the U.S.C section? `U\.*S\.*C\.*`

Reply:
Good catch, @ccostino! We may or may not have the ".", and if we have it we have only one of them, so it should be `U\.?S\.?C\.?`. Ditto for CFRs.